src/agents/stepping.jl

```diff
@@ -0,0 +1,98 @@
+struct ABMStepper{X, C, M, S, O}
+    ac::C
+    am::M
+    as::S
+    offset::O
+    scheduler::X
+    pos::Observable
+    colors::Observable
+    sizes::Observable
+    markers::Observable
+end
+
+Base.show(io::IO, ::ABMStepper) =
+println(io, "Helper structure for stepping and updating the plot of an agent based model. ",
+"It is outputted by `abm_plot` and can be used in `Agents.step!`.")
+
+"Initialize the abmstepper and the plotted observables. return the stepper"
+function abm_init_stepper_and_plot!(ax, model;
+        ac = JULIADYNAMICS_COLORS[1],
+        as = 1,
+        am = :circle,
+        scheduler = model.scheduler,
+        offset = nothing,
+        equalaspect = true,
+        scatterkwargs = NamedTuple(),
+    )
+
+    o, e = modellims(model) # TODO: extend to 3D
+    @assert length(o) == 2 "At the moment only 2D spaces can be plotted."
+    # TODO: once graph plotting is possible, this will be adjusted
+    @assert typeof(model.space) <: Union{Agents.ContinuousSpace, Agents.DiscreteSpace}
+
+    ids = scheduler(model)
+    colors  = ac isa Function ? Observable(to_color.([ac(model[i]) for i ∈ ids])) : to_color(ac)
+    sizes   = as isa Function ? Observable([as(model[i]) for i ∈ ids]) : as
```

Comment:
If `sizes = as`, then when sending it into `ABMStepper` on line 53 this will fail right? same with `colors` and `markers`.

Reply:
Oh yeah lol that was a simple oversight, the struct needs to parameterized on the final fields as well. Extremely simple fix